conftest.py

```diff
@@ -179,13 +180,20 @@ def _get_sensor_reading():
 def set_sensor_name(make_valid_request):
     def _get_sensor_name(name:str):
         log.info("Set sensor name to %s", name)
+        if not name:
```

Comment:
This is good thinking but you need these functions to be able to send invalid data to sensor. Adding a condition here that raises a ValueError prevents you from doing that.